simulation/groups_interface.py

```diff
@@ -4,117 +4,44 @@
 
 
 from abc import abstractmethod, ABCMeta
+import random
 
 
 class PopulationGroup(metaclass=ABCMeta):
     """
     Abstract class for specific population groups based on SIR model.
     """
+
     @abstractmethod
-    def __init__(self, x: float, y: float, label: str, inf_dist: float,
-                 infection_prob: float, recover_prob: float, death_prob: float) -> None:
+    def __init__(self, x: float, y: float, label: str, **kwargs) -> None:
```

Comment:
Why we have **kwargs here?
This is implicit pass of parameters, it is always better to be explicit (better readability).
Anyway, I guess it is here because you wanted to keep the same parameters as in other class implementations.
But you could define different parameters for these methods without any issues. Child class could easily override parent method with own parameters, and this is ok.